### PR TITLE
Use tc-lib-pulse in place of pulse listener and pulse publisher

### PR DESCRIPTION
**Pull request by Biboswan**

### CODE REVIEW

**Comment by Biboswan** [comment]
@djmitche

File: config.yml [hunk 1]
```diff
@@ -42,11 +42,21 @@ defaults:
...
+      contact:                 !env PULSE_CONTACT
+      rootUrl:                 !env TASKCLUSTER_ROOT_URL
+      credentials:
+        clientId:              !env TASKCLUSTER_CLIENT_ID
+        accessToken:           !env TASKCLUSTER_ACCESS_TOKEN
+    client:
+      namespace:               !env PULSE_NAMESPACE
+      recycleInterval:         !env PULSE_CLIENT_RECYCLE_INTERVAL
+      retirementDelay:         !env PULSE_CLIENT_RETIREMENT_DELAY
+      minReconnectionInterval: !env PULSE_CLIENT_MIN_RECONNECTION_INTERVAL
```
Comment (djmitche) [inline config.yml]: I think this is too much configuration :)

The only things that are really necessary here are the namespace, expiresAfter, and contact.  The client options can be left at their default values, and the others are available elsewhere in the config.  So the result would be

```js
pulse:
  namespace: !env PULSE_NAMESPACE
  contact: !env PULSE_CONTACT
  expiresAfter: !env PULSE_EXPIRES_AFTER
```

Then when constructing the client, use `cfg.taskcluster` for the credentials/rootUrl, and pass `cfg.pulse.namespace` as necessary.

File: src/handler.js [hunk 2]
```diff
@@ -3,37 +3,36 @@ const _ = require('lodash');
...
+      queueEvents.taskException(`route.${routePrefix}.#.on-any.#`),
+    ];
   }
 
   async listen() {
-    if (!this.testing) {
-      await this.listener.connect();
-    }
-    await this.listener.resume();
+    this.pq = await consume({client:this.pulseClient, bindings:this.bindings,
```
[the remark this answers is not in the document]
Comment (Biboswan) [reply src/handler.js]: This space problem is with me in every bug :)

**Comment by Biboswan** [comment]
made the changes

**Comment by djmitche** [comment]
I'll work on deploying this tomorrow morning

File: src/handler.js [hunk 3]
```diff
@@ -3,37 +3,38 @@ const _ = require('lodash');
...
-    if (!this.testing) {
-      await this.listener.connect();
-    }
-    await this.listener.resume();
+    this.pq = await consume({
+      client: this.pulseClient,
+      bindings: this.bindings,
+      queueName: this.queueName,
+    },
+    (message) => this.monitor.timedHandler('notification', this.onMessage(message)));
```
Comment (owlishDeveloper) [inline src/handler.js]: @Biboswan heya, I was working on a similar PR for tc-gh, and used this one as my guide. This line here caught my attention... Is `this.onMessage(message)` correct?

Comment (Biboswan) [reply src/handler.js]: Hi. Yeah that's correct. Otherwise hardly any tests would have passed. But from irc I heard `timehandler` has some issue.

**Pull request merged**